@adamsitnik Thanks for submitting a PR!

I played with your changes, and it seems to break the ability to target a specific version of `Microsoft.PowerShell.SDK`. For example:
```
PS: benchmarks> $env:PERF_TARGET_VERSION = '7.2.0-preview.5'
PS: benchmarks> dotnet run -c release -f net6.0 --filter *using* --keepFiles

C:\arena\source\PowerShell\test\perf\benchmarks\powershell-perf.csproj : error NU1202: Package Microsoft.PowerShell.Commands.Diagnostics 7.2.0-preview.5 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Microsoft.PowerShell.Commands.Diagnostics 7.2.0-preview.5 supports: net6.0 (.NETCoreApp,Version=v6.0)
C:\arena\source\PowerShell\test\perf\benchmarks\powershell-perf.csproj : error NU1202: Package Microsoft.PowerShell.Commands.Management 7.2.0-preview.5 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Microsoft.PowerShell.Commands.Management 7.2.0-preview.5 supports: net6.0 (.NETCoreApp,Version=v6.0)
C:\arena\source\PowerShell\test\perf\benchmarks\powershell-perf.csproj : error NU1202: Package Microsoft.PowerShell.Commands.Utility 7.2.0-preview.5 is not compatible with netcoreapp3.1 (.NETCoreApp,Version=v3.1). Package Microsoft.PowerShell.Commands.Utility 7.2.0-preview.5 supports: net6.0 (.NETCoreApp,Version=v6.0)

...
```
In this case, `7.2.0-preview.5` is a preview version built against `net6.0`.
I tried to make it work but my efforts didn't go anywhere. Tried `--envVars PERF_TARGET_VERSION:7.2.0-preview.5` but it turned out the environment variable specified this way doesn't affect the build phase of the auto-generated project.

To us, the ability to compare against different PowerShell SDK is more important than running benchmarks against different .NET runtimes. To some extent, running benchmarks with different PowerShell SDKs targeting the same `net6.0` runtime help us to eliminate the runtime difference, and only comparing with the changes in PowerShell.

BTW, the building of the auto-generated project is still biting us because we cannot specify `--no-dependencies` for the first try
> // start dotnet build -c Release /p:DebugType=portable --no-restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true in C:\arena\source\PowerShell\test\perf\benchmarks\bin\release\net6.0\Job-DYFHWE
// command took 58.65s and exited with 1

I think it makes sense to try `--no-dependencies` first, and fall back to a full build if that failed. I can submit a PR to BenchmarkDotNet to make that change if you agree that's the way to go.

--- a/test/perf/benchmarks/README.md
+++ b/test/perf/benchmarks/README.md
@@ -24,12 +24,12 @@ This folder contains micro benchmarks that test the performance of PowerShell En
 You can run the benchmarks directly using `dotnet run` in this directory:
 1. To run the benchmarks in Interactive Mode, where you will be asked which benchmark(s) to run:
    ```
-   dotnet run -c release
+   dotnet run -c Release
    ```
 
 2. To list all available benchmarks ([read more](https://github.com/dotnet/performance/blob/main/docs/benchmarkdotnet.md#Listing-the-Benchmarks)):
    ```
-   dotnet run -c release --list [flat/tree]
+   dotnet run -c Release --list [flat/tree]
    ```
 
 3. To filter the benchmarks using a glob pattern applied to `namespace.typeName.methodName` ([read more](https://github.com/dotnet/performance/blob/main/docs/benchmarkdotnet.md#Filtering-the-Benchmarks)]):

--- a/test/perf/benchmarks/powershell-perf.csproj
+++ b/test/perf/benchmarks/powershell-perf.csproj
@@ -24,8 +24,8 @@
     <PerfTargetVersion>$(PERF_TARGET_VERSION)</PerfTargetVersion>
 
     <!-- Test.Common.props sets TargetFramework to net6.0, we need to clear this value to be able to target multiple TFMs.
-         But only when PERF_TARGET_VERSION was not specified. When it it is specified, a specific version is being benchmarked
-         And this version can be benchmarked only against single runtime version. -->
+         But only when PERF_TARGET_VERSION was not specified. When it is specified, a specific version is being benchmarked
+         and this version can be benchmarked only against single runtime version (because Microsoft.PowerShell.SDK targets a single TFM). -->
     <TargetFramework Condition="'$(PerfTargetVersion)' == ''"></TargetFramework>
     <TargetFrameworks Condition="'$(TargetFramework)' == ''">netcoreapp2.1;net5.0;net6.0</TargetFrameworks>
 

--- a/test/perf/benchmarks/powershell-perf.csproj
+++ b/test/perf/benchmarks/powershell-perf.csproj
@@ -16,16 +16,19 @@
     <DebugType>portable</DebugType>
     <DebugSymbols>true</DebugSymbols>
 
-    <!-- Test.Common.props sets TargetFramework to net6.0, we need to clear this value to be able to target multiple TFMs -->
-    <TargetFramework></TargetFramework>
-    <TargetFrameworks>netcoreapp2.1;net5.0;net6.0</TargetFrameworks>
-    
     <!-- To run benchmarks targeting a specific package version, set the version to 'PERF_TARGET_VERSION' as an environment variable.
      Do not try passing in the value using '/property:' at command line, because
        1. 'dotnet run' doesn't respect '/property:' arguments;
        2. BenchmarkDotnet generates temporary project files that reference to this .csproj file,
           and '/property:' arguments won't be forwarded when building those temp projects. -->
     <PerfTargetVersion>$(PERF_TARGET_VERSION)</PerfTargetVersion>
+
+    <!-- Test.Common.props sets TargetFramework to net6.0, we need to clear this value to be able to target multiple TFMs.
+         But only when PERF_TARGET_VERSION was not specified. When it it is specified, a specific version is being benchmarked
+         And this version can be benchmarked only against single runtime version. -->
+    <TargetFramework Condition="'$(PerfTargetVersion)' == ''"></TargetFramework>
+    <TargetFrameworks Condition="'$(TargetFramework)' == ''">netcoreapp2.1;net5.0;net6.0</TargetFrameworks>
+
     <PerfTargetVersion Condition="'$(PerfTargetVersion)' == '' AND '$(TargetFramework)' == 'net5.0'">7.1.3</PerfTargetVersion>
     <PerfTargetVersion Condition="'$(PerfTargetVersion)' == '' AND '$(TargetFramework)' == 'netcoreapp2.1'">6.2.7</PerfTargetVersion>
   </PropertyGroup>

--- a/test/perf/perf.psm1
+++ b/test/perf/perf.psm1
@@ -76,7 +76,7 @@ function Start-Benchmarking
             if ($List) { $runArgs += '--list', $List }
             if ($KeepFiles) { $runArgs += "--keepFiles" }
 
-            dotnet run -c release -f $TargetFramework --filter $Filter --artifacts $Artifacts --envVars POWERSHELL_TELEMETRY_OPTOUT:1 $runArgs
+            dotnet run -c Release -f $TargetFramework --filter $Filter --artifacts $Artifacts --envVars POWERSHELL_TELEMETRY_OPTOUT:1 $runArgs
 
             if (Test-Path $Artifacts) {
                 Write-Log -message "`nBenchmark artifacts can be found at $Artifacts"

